graphql-dgs-spring-graphql/src/main/kotlin/com/netflix/graphql/dgs/springgraphql/DgsGraphQLSourceBuilder.kt

```diff
@@ -0,0 +1,164 @@
+        graphQlConfigurers = this.graphQlConfigurers.andThen(configurer)
+        return this
+    }
+
+    override fun build(): GraphQlSource {
+        return ReloadableGraphQLSource(reload(), this, reloadSchemaIndicator)
+    }
+
+    fun reload(): GraphQlSource {
+        var schema: GraphQLSchema = dgsSchemaProvider.schema(schemaResources = schemaResources)
+        val schemaTransformer = SchemaTransformer()
+        typeVisitorsToTransformSchema.forEach {
+            schemaTransformer.transform(schema, it)
+        }
+
+        schema = this.applyTypeVisitors(schema)
+        var builder = GraphQL.newGraphQL(schema)
+        builder.defaultDataFetcherExceptionHandler(DataFetcherExceptionResolver.createExceptionHandler(this.exceptionResolvers))
+        if (!instrumentations.isEmpty()) {
+            builder = builder.instrumentation(ChainedInstrumentation(this.instrumentations))
+        }
+
+        graphQlConfigurers.accept(builder)
+
+        return object : GraphQlSource {
+            val graphql = builder.build()
+
+            override fun graphQl(): GraphQL {
+                return graphql
+            }
+
+            override fun schema(): GraphQLSchema {
+                return schema
+            }
+        }
+    }
+
+    private fun applyTypeVisitors(schema: GraphQLSchema): GraphQLSchema {
+        val outputCodeRegistry =
+            GraphQLCodeRegistry.newCodeRegistry(schema.codeRegistry)
+
+        val vars: MutableMap<Class<*>, Any> = HashMap(2)
+        vars[GraphQLCodeRegistry.Builder::class.java] = outputCodeRegistry
+        vars[TypeVisitorHelper::class.java] = TypeVisitorHelper.create(schema)
+
+        val visitorsToUse: MutableList<GraphQLTypeVisitor> = ArrayList(this.typeVisitors)
+        visitorsToUse.add(ContextDataFetcherDecorator.createVisitor(this.subscriptionExceptionResolvers))
+
+        SchemaTraverser().depthFirstFullSchema(visitorsToUse, schema, vars)
+        return schema.transformWithoutTypes { builder: BuilderWithoutTypes ->
+            builder.codeRegistry(
+                outputCodeRegistry
+            )
+        }
+    }
+
+    override fun schemaResources(vararg resources: Resource?): SchemaResourceBuilder {
+        schemaResources.plus(listOf(*resources))
+        return this
+    }
+
+    override fun configureTypeDefinitions(configurer: TypeDefinitionConfigurer): SchemaResourceBuilder {
+        this.typeDefinitionConfigurers.add(configurer)
+        return this
+    }
+
+    override fun configureRuntimeWiring(configurer: RuntimeWiringConfigurer): SchemaResourceBuilder {
+        this.runtimeWiringConfigurers.add(configurer)
+        return this
+    }
+
+    override fun defaultTypeResolver(typeResolver: TypeResolver): SchemaResourceBuilder {
+        this.typeResolver = typeResolver
+        return this
+    }
+
+    override fun inspectSchemaMappings(reportConsumer: Consumer<SchemaReport>): SchemaResourceBuilder {
+        this.schemaReportConsumer = reportConsumer
```

Comment:
The SchemaReport is also worth a mention. What's required to run it is a `GraphQLSchema` and a `RuntimeWiring` and it's currently run from `DefaultSchemaResourceGraphQlSourceBuilder`, and since you are replacing that, you'll need to invoke it from this builder implementation. 

As part of https://github.com/spring-projects/spring-graphql/issues/878, I've added a protected method on `AbstractGraphQlSourceBuilder` that lets you hook in at the point where `GraphQL.Builder` is customized and that is where we invoke the schema report if the application has explicitly provided a `SchemaReport` consumer.  

If you look at the commit attached to that issue, you can see how it's done. DGS can do something similar but would need to obtain the resulting `RuntimeWiring` from `DgsSchemaProvider` somehow. Mainly what we need from the `RuntimeWiring` is the map of DataFetchers's if it helps to know that.